source/agora/node/Node.d

```diff
@@ -226,7 +226,10 @@ public class Node : API
     /// GET: /node_info
     public override NodeInfo getNodeInfo () pure nothrow @safe @nogc
     {
-        return this.network.getNetworkInfo();
+        // NetworkInfo only brings network-related information.
+        NodeInfo node_info = this.network.getNetworkInfo();
```

Comment:
If we get a NodeInfo object through the `getNetworkInfo` function, why don't you change the name of the function into `getNodeInfo` also?

Reply:
Like the annotation written on the 229th line, getNetworkInfo only brings network information, and NodeInfo is a higher concept than NetworkInfo, so getNetNetworkInfo in the network uses its function as it is.

Reply:
And since the assigned struct is `NodeInfo`, I assigned the variable name to `node_info`

Reply:
It's still misleading to have a function named `getNetworkInfo` while returning a different type.

I would suggest doing something like this:

```D
NodeInfo node_info;
this.network.fillNetworkInfo(node_info.state, node_info.addresses);
node_info.is_validator = this.config.node.is_validator;
```

That means changing the function name and its code.